config/src/main/java/com/typesafe/config/impl/ConfigNodeSimpleValue.java

```diff
@@ -16,4 +17,21 @@
     protected Collection<Token> tokens() {
         return Collections.singletonList(token);
     }
+
+    protected Token token() { return token; }
+
+    protected AbstractConfigValue value() {
+        if (Tokens.isValue(token))
+            return Tokens.getValue(token);
+        else if (Tokens.isUnquotedText(token))
+            return new ConfigString.Unquoted(token.origin(), Tokens.getUnquotedText(token));
+        else if (Tokens.isSubstitution(token)) {
+            List<Token> expression = Tokens.getSubstitutionPathExpression(token);
+            Path path = PathParser.parsePathExpression(expression.iterator(), token.origin());
+            boolean optional = Tokens.getSubstitutionOptional(token);
+
+            return new ConfigReference(token.origin(), new SubstitutionExpression(path, optional));
+        }
+        return null;
```

Comment:
Yeah, I don't think this `return null` should ever be reached, it should probably throw a `BugOrBrokenException`.